dev_apps/music-nga/js/endpoint.js

```diff
@@ -276,7 +276,14 @@ function getAlbum(filePath) {
     var album = song.metadata.album;
 
     return new Promise((resolve) => {
-      Database.enumerateAll('metadata.album', album, 'next', songs => resolve(songs));
+      Database.enumerateAll('metadata.album', album, 'next', (songs) => {
+        songs.sort((a, b) => {
+          return (a.metadata.discnum - b.metadata.discnum) ||
+                 (a.metadata.tracknum - b.metadata.tracknum);
+        });
```

Comment:
This doesn't belong in the front-end; ordering guarantees should probably be a part of the database backend. I think it'd make more sense in `js/db.js`.